spec/lib/tasks/after_tasks.rake_spec.rb

```diff
@@ -170,21 +170,14 @@
 end
 
 describe "rake After:clean_up_noncanonical_categories" do
-  let(:noncanonical_category_tag) { Category.create(name: "Borked category tag", canonical: false) }
-  let(:canonical_category_tag) { Category.find_or_create_by(name: ArchiveConfig.CATEGORY_GEN_TAG_NAME, canonical: true) }
-  let(:work_with_noncanonical_categ) { create(:work) }
-  let(:work_with_canonical_and_noncanonical_categs) { create(:work) }
-
-  before do
-    work_with_noncanonical_categ.categories = [noncanonical_category_tag]
-    work_with_noncanonical_categ.save!
-    work_with_canonical_and_noncanonical_categs.categories = [noncanonical_category_tag, canonical_category_tag]
-    work_with_canonical_and_noncanonical_categs.save!
-  end
+  # Must use let!() for all of these here, as these are preconditions to running the task
```

Comment:
We use `let!` quite often in the test suite, I think you can leave out this comment.